File: examples/r/aoc.R
library("purrr")

input <- purrr::map(readLines("input.txt"), strtoi)

getSolutionPart1 <- function() {
    purrr::reduce(input, `+`)
}

getSolutionPart2 <- function() {
    purrr::reduce(input, `*`)
}

part <- Sys.getenv("part", unset = "part1")

if (part == "part1") {
    print (getSolutionPart1())
} else if (part == "part2") {
    print (getSolutionPart2())
} else {
    print (paste("Unknown part", part, sep=" "))
}

File: examples/python/aoc.py
from functools import reduce
from operator import mul
from os import environ


def getSolutionPart1(input_list):
    return sum(input_list)


def getSolutionPart2(input_list):
    return reduce(mul, input_list)


with open('input.txt') as f:
    file_input = [int(x) for x in f.readlines()]

print('Python')
part = environ.get('part')

if part == 'part2':
    print(getSolutionPart2(file_input))
else:
    print(getSolutionPart1(file_input))
